pkg/kubernetes/kubernetes.go

```diff
@@ -38,7 +39,8 @@ const (
 type CloseWatchKubeConfig func() error
 
 type Kubernetes struct {
-	manager *Manager
+	manager          *Manager
+	podClientFactory func(namespace string) (corev1client.PodInterface, error)
```

Comment:
Wouldn't these changes in here cause the divergence from upstream?. I think, in this repository we shouldn't allow any changes touching these packages such as kubernetes/, mcp/, etc.

My suggestion is to add first in upstream and simply use it in downstream. Downstream repository should touch only `ocp/` directory.

Reply:
/hold